proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,448 @@
+
+This proposal puts a possible set of _area teams_ to form with the specified areas of responsibility. One notable point is that LLVM (`:/llvm`) had 945 unique authors contribute over the 12 month period used for statistics. This is far more than any other part of the project, and it justified splitting LLVM into two teams, which are separated as the _LLVM Backend_ and _LLVM Middle-end_. Clang had 706 unique authors, which is also significantly higher than other sub-projects. To represent different parts of the Clang community the _Clang_ and _Clang Tooling_ teams are separated below.
+
+* LLVM Backend (CodeGen, MC, Target)
+* LLVM Middle-end (Everything not covered by LLVM Backend)
+* Clang (All the `clang` sub-project except tooling and format libraries and tools)
+* Clang Tooling (`clang-tools-extra` and the tooling and format libraries from `clang`)
+* C/C++ Runtime Libraries (`libcxx`, `libc`, `libunwind`, `libcxxabi`)
+* Compiler Runtimes (`compiler-rt`, `openmp`)
+* Flang (`flang`)
+* MLIR (`mlir`)
+* LLDB (`lldb`)
+* Binary tools (`lld`, `bolt`)
+* Incubator (`circt`, `torch-mlir`, `Polygeist`, `clangir`, etc)
+* Project Infrastructure (`zorg`, `utils`, `llvm-test-suite`, CI/CD, hosting, etc)
+* Community (Discord, Discourse, etc)
+
+##### Community & Infrastructure Area Teams
+
+The _Community_ and _Infrastructure_ _area teams_ have different responsibilities from the other _area teams_ because they aren’t restricted to specific software components of the LLVM project. Their goals are more holistic to provide for the needs of the whole project and wider community.
+
+In addition to a team chair and secretary, both the _community_ and _infrastructure area teams_ will select one member of their team to act as a liaison with the LLVM Foundation. The liaisons will be responsible for managing the relationship between their _area team_ and the employees and directors of the LLVM Foundation. The liaison may be any member of the _area team_ including the chair or secretary.
+
+##### Voting by Area
+
+Every _voting contributor_ is eligible to vote for the project-wide _Community_ and _Infrastructure_ teams. Eligibility for voting on other _area teams_ is determined by _meaningful contributions_ to the areas those teams cover. For example, in order to vote on the **LLVM Backend** _area team_, a contributor would need to contribute to the LLVM `CodeGen`, `MC`, or `Target` libraries, or to vote on the members of the **Flang** team, a contributor would need to contribute to the `flang` project. There is no limit to the number of _area teams_ a contributor can vote on if they have made contributions to the relevant areas.
```

Comment:
I still think it's going to be too difficult to determine eligibility for each area team.  I would recommend that we just have a single pool of voters.  Given that one member from each area team will serve on the project council, and will make decisions that affect the whole project, I think it still useful to have people vote in areas they don't actively contribute to of they want.  If we are really concerned about this, maybe we could limit the number of area teams people can vote for (e.g. You can only vote in 5 area team elections).   This kind of restriction is going to be much easier to implement than trying to group voters based on their contributions.